components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/model/template/service/BallerinaOperation.java

```diff
@@ -110,14 +114,47 @@ public BallerinaOperation buildContext(Operation operation, ExtendedAPI api) thr
                     this.isSecured = false;
                 }
             });
-            //set resource level request interceptors
+
+            /*
+            Set resource level request interceptors and set the ballerina module to be imported if specified.
+            */
             Optional<Object> requestInterceptor = Optional.ofNullable(extensions
                     .get(OpenAPIConstants.REQUEST_INTERCEPTOR));
-            requestInterceptor.ifPresent(value -> this.requestInterceptor = value.toString());
-            //set resource level response interceptors
+            if (requestInterceptor.toString().contains(OpenAPIConstants.MODULE_STATEMENT_SEPARATOR)) {
```

Comment:
You can move this to a method and use it for both request and response interceptor.